code/modules/mob/living/superior_animal/giant_spider/giant_spider.dm

```diff
@@ -168,32 +167,40 @@
 					spawn(50)
 						if(busy == SPINNING_COCOON)
 							if(cocoon_target && istype(cocoon_target.loc, /turf) && get_dist(src,cocoon_target) <= 1)
-								var/obj/effect/spider/cocoon/C = new(cocoon_target.loc)
-								var/large_cocoon = 0
-								C.pixel_x = cocoon_target.pixel_x
-								C.pixel_y = cocoon_target.pixel_y
-								for(var/mob/living/M in C.loc)
-									if(istype(M, /mob/living/superior_animal/giant_spider))
+								var/obj/effect/spider/cocoon/C = locate() in cocoon_target.loc
+								var/large_cocoon
+								var/turf/targetTurf = cocoon_target.loc
+
+								for(var/obj/O in targetTurf)
+									if (O.anchored)
+										continue
+									if (istype(O, /obj/item))
+									else if (istype(O, /obj/structure) || istype(O, /obj/machinery))
+										large_cocoon = 1
+									else
+										continue
+
+									C = C || new(targetTurf)
+									O.loc = C
+
+								for(var/mob/living/M in targetTurf)
+									if((M.stat == CONSCIOUS) || istype(M, /mob/living/superior_animal/giant_spider))
 										continue
 									large_cocoon = 1
-									fed++
-									src.visible_message(SPAN_WARNING("\The [src] sticks a proboscis into \the [cocoon_target] and sucks a viscous substance out."))
+
+									if ((M.getBruteLoss() < 150) && istype(M, /mob/living/carbon))
```

Comment:
I am really not keen on this. The hardcoded check for 150, the direct application of 100 damage, it smells like bad code

Rather than directly applying damage, it would make more sense to drain blood, That's at least something more flexible and gracefully handled.

The way this is implemented doesn't sit right with me at all